### PR TITLE
chore: remove true comparision

### PR DESCRIPTION
**Pull request by TropicalDog17**
## Description

Remove instances of redundant logical conditions that alway result in true.

## Summary by CodeRabbit

- **Refactor**
	- Simplified file path handling in the CLI application for better efficiency and readability.
	- Improved the logic for decoding hexadecimal nibbles by removing redundant conditions.

### CODE REVIEW

File: mod/cli/pkg/commands/jwt/jwt.go [hunk 1]
```diff
@@ -117,16 +117,14 @@ func getFilePath(cmd *cobra.Command, path string) (string, error) {
...
-		specifiedFilePath = filepath.Join(
-			clientCtx.HomeDir+"/config/", DefaultSecretFileName,
-		)
+	clientCtx, ok := cmd.Context().
+		Value(client.ClientContextKey).(*client.Context)
+	if !ok {
+		return "", ErrNoClientCtx
 	}
+	specifiedFilePath = filepath.Join(
+		clientCtx.HomeDir+"/config/", DefaultSecretFileName,
```
Comment (archbear) [inline mod/cli/pkg/commands/jwt/jwt.go:126]: nit: filepath.Join appends the backslashes correctly according to os spec so hardcoding /config/ isn't necessary if i remember correctly

**Review by archbear — APPROVED** [review]
nit otherwise lgtm

File: mod/primitives/pkg/hex/nibble.go [hunk 2]
```diff
@@ -28,11 +28,11 @@ import (
 func decodeNibble(in byte) uint64 {
 	// uint64 conversion here is safe
 	switch {
-	case in >= '0' && in <= '9' && in >= hexBaseOffset:
```
Comment (itsdevbear) [inline mod/primitives/pkg/hex/nibble.go:31]: what's this change?

File: mod/cli/pkg/commands/jwt/jwt.go [hunk 3]
```diff
@@ -117,16 +118,14 @@ func getFilePath(cmd *cobra.Command, path string) (string, error) {
 	// If no path is specified, try to get the cosmos client context and use
 	// the configured home directory to write the secret to the default file
 	// name.
-	if specifiedFilePath == "" {
```
Comment (itsdevbear) [inline mod/cli/pkg/commands/jwt/jwt.go:120]: i don't think this is logically equivalent

Comment (ocnc2) [reply mod/cli/pkg/commands/jwt/jwt.go:120]: is it not?

File: mod/primitives/pkg/hex/nibble.go (hunk 2 above)
Comment (TropicalDog17) [reply mod/primitives/pkg/hex/nibble.go:31]: hexBaseOffset is already '0' so no need for comparison here. Same with the other `if` statements

File: mod/cli/pkg/commands/jwt/jwt.go (hunk 3 above)
Comment (TropicalDog17) [reply mod/cli/pkg/commands/jwt/jwt.go:120]: we already early return if `specifiedFilePath` is not empty [here](https://github.com/berachain/beacon-kit/blob/056a0a5f53a6898672ad44048dd997693a8a66e6/mod/cli/pkg/commands/jwt/jwt.go#L114) so no need to check again.

**Review by ocnc2 — APPROVED** [review]

File: mod/primitives/pkg/hex/nibble.go (hunk 2 above)
Comment (ocnc2) [reply mod/primitives/pkg/hex/nibble.go:31]: you're right. wanna add a brief comment clarifying that this would never underflow?

Comment (ocnc2) [reply mod/primitives/pkg/hex/nibble.go:31]: or maybe above each return, add a comment containing the expected range of the result. 

ex. `in - hexBaseOffset` would result in a value in 0-9, `in - hexAlphaOffsetUpper` would result in a value in 10-15, etc.

**Review by itsdevbear — APPROVED** [review]
yup I agree with all points, 

@TropicalDog17 just add a comment for what ocnc2 and i'm good to merge

**Review by ocnc2 — APPROVED** [review]
lgtm thanks @TropicalDog17!

**Pull request merged**